Serilog dropped cross-build for netcoreapp

**Description**
A dependency resolution produces a lot of warnings. 
```log
....csproj : warning NU1603: Microsoft.Extensions.Configuration 2.1.1 depends on Microsoft.Extensions.Configuration.Abstractions (>= 2.1.1) but Microsoft.Extensions.Configuration.Abstractions 2.1.1 was not found. An approximate best match of Microsoft.Extensions.Configuration.Abstractions 3.0.0 was resolved.
....csproj : warning NU1603: Microsoft.Extensions.Hosting.Abstractions 2.1.0 depends on Microsoft.Extensions.Configuration.Abstractions (>= 2.1.0) but Microsoft.Extensions.Configuration.Abstractions 2.1.0 was not found. An approximate best match of Microsoft.Extensions.Configuration.Abstractions 3.0.0 was resolved.
....csproj : warning NU1603: Microsoft.Extensions.Hosting.Abstractions 2.2.0 depends on Microsoft.Extensions.Configuration.Abstractions (>= 2.2.0) but Microsoft.Extensions.Configuration.Abstractions 2.2.0 was not found. An approximate best match of Microsoft.Extensions.Configuration.Abstractions 3.0.0 was resolved.
....csproj : warning NU1603: Microsoft.Extensions.Options.ConfigurationExtensions 2.1.0 depends on Microsoft.Extensions.Configuration.Abstractions (>= 2.1.0) but Microsoft.Extensions.Configuration.Abstractions 2.1.0 was not found. An approximate best match of Microsoft.Extensions.Configuration.Abstractions 3.0.0 was resolved.
....csproj : warning NU1603: Microsoft.Extensions.Options.ConfigurationExtensions 2.0.0 depends on Microsoft.Extensions.Configuration.Binder (>= 2.0.0) but Microsoft.Extensions.Configuration.Binder 2.0.0 was not found. An approximate best match of Microsoft.Extensions.Configuration.Binder 2.0.1 was resolved.
....csproj : warning NU1603: Microsoft.Extensions.Options.ConfigurationExtensions 2.1.0 depends on Microsoft.Extensions.Configuration.Binder (>= 2.1.0) but Microsoft.Extensions.Configuration.Binder 2.1.0 was not found. An approximate best match of Microsoft.Extensions.Configuration.Binder 2.1.1 was resolved.
....csproj : warning NU1603: Microsoft.Extensions.Logging 3.0.0 depends on Microsoft.Extensions.Configuration.Binder (>= 3.0.0) but Microsoft.Extensions.Configuration.Binder 3.0.0 was not found. An approximate best match of Microsoft.Extensions.Configuration.Binder 3.0.1 was resolved.
```

**What happens?/How to reproduce**
Require `Serilog.AspNetCore@4.0.0` -> it dropped conditional build for `netcoreapp3.1` (and replaced it with `netstandard2.1` - which is not the same). This adds a dependency on `Microsoft.AspNetCore.Hosting.Abstractions` which is a package from `netcoreapp2.0`.

**Expected behavior**
For `netcoreapp3.0` and higher do not depend on `Microsoft.AspNetCore.*@2.x` packages.

**Additional context**
I can prepare a PR, if you like.

--- a/src/Serilog.AspNetCore/Serilog.AspNetCore.csproj
+++ b/src/Serilog.AspNetCore/Serilog.AspNetCore.csproj
@@ -32,26 +32,32 @@
     <PackageReference Include="Serilog.Sinks.Debug" Version="2.0.0" />
     <PackageReference Include="Serilog.Settings.Configuration" Version="3.1.0" />
     <PackageReference Include="Serilog.Formatting.Compact" Version="1.1.0" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.8" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.8" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.2.0" />
   </ItemGroup>
   
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.1'">
     <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.2.0" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.8" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.8" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
   </ItemGroup>
 
 </Project>

--- a/src/Serilog.AspNetCore/Serilog.AspNetCore.csproj
+++ b/src/Serilog.AspNetCore/Serilog.AspNetCore.csproj
@@ -4,7 +4,7 @@
     <Description>Serilog support for ASP.NET Core logging</Description>
     <VersionPrefix>4.0.1</VersionPrefix>
     <Authors>Microsoft;Serilog Contributors</Authors>
-    <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.1;net5.0</TargetFrameworks>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <AssemblyOriginatorKeyFile>../../assets/Serilog.snk</AssemblyOriginatorKeyFile>
@@ -32,9 +32,26 @@
     <PackageReference Include="Serilog.Sinks.Debug" Version="2.0.0" />
     <PackageReference Include="Serilog.Settings.Configuration" Version="3.1.0" />
     <PackageReference Include="Serilog.Formatting.Compact" Version="1.1.0" />
-    <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.0" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.8" />
     <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.8" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
+  </ItemGroup>
+  
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.1'">
+    <PackageReference Include="Microsoft.AspNetCore.Hosting.Abstractions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
   </ItemGroup>
 
 </Project>

--- a/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
+++ b/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
@@ -16,7 +16,6 @@ using System;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Logging;
-using Serilog.Core;
 using Serilog.Extensions.Hosting;
 using Serilog.Extensions.Logging;
 
@@ -33,8 +32,8 @@ namespace Serilog
         /// <param name="builder">The web host builder to configure.</param>
         /// <param name="logger">The Serilog logger; if not supplied, the static <see cref="Serilog.Log"/> will be used.</param>
         /// <param name="dispose">When true, dispose <paramref name="logger"/> when the framework disposes the provider. If the
-        /// logger is not specified but <paramref name="dispose"/> is true, the <see cref="Log.CloseAndFlush()"/> method will be
-        /// called on the static <see cref="Log"/> class instead.</param>
+        /// logger is not specified but <paramref name="dispose"/> is true, the <see cref="Serilog.Log.CloseAndFlush()"/> method will be
+        /// called on the static <see cref="Serilog.Log"/> class instead.</param>
         /// <param name="providers">A <see cref="LoggerProviderCollection"/> registered in the Serilog pipeline using the
         /// <c>WriteTo.Providers()</c> configuration method, enabling other <see cref="ILoggerProvider"/>s to receive events. By
         /// default, only Serilog sinks will receive events.</param>
@@ -78,8 +77,8 @@ namespace Serilog
         /// The logger will be shut down when application services are disposed.
         /// </remarks>
         /// <param name="builder">The web host builder to configure.</param>
-        /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="Logger" />.</param>
-        /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Log.Logger"/>.</param>
+        /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="Serilog.Core.Logger" />.</param>
+        /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Serilog.Log.Logger"/>.</param>
         /// <param name="writeToProviders">By default, Serilog does not write events to <see cref="ILoggerProvider"/>s registered through
         /// the Microsoft.Extensions.Logging API. Normally, equivalent Serilog sinks are used in place of providers. Specify
         /// <c>true</c> to write events to all providers.</param>

--- a/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
+++ b/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
@@ -77,7 +77,7 @@ namespace Serilog
         /// The logger will be shut down when application services are disposed.
         /// </remarks>
         /// <param name="builder">The web host builder to configure.</param>
-        /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="ILogger" />.</param>
+        /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="Logger" />.</param>
         /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Log.Logger"/>.</param>
         /// <param name="writeToProviders">By default, Serilog does not write events to <see cref="ILoggerProvider"/>s registered through
         /// the Microsoft.Extensions.Logging API. Normally, equivalent Serilog sinks are used in place of providers. Specify

--- a/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
+++ b/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
@@ -13,11 +13,11 @@
 // limitations under the License.
 
 using System;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Logging;
-using Serilog.Extensions.Logging;
-using Microsoft.Extensions.DependencyInjection;
 using Serilog.Extensions.Hosting;
+using Serilog.Extensions.Logging;
 
 namespace Serilog
 {
@@ -77,7 +77,7 @@ namespace Serilog
         /// The logger will be shut down when application services are disposed.
         /// </remarks>
         /// <param name="builder">The web host builder to configure.</param>
-        /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="Logger" />.</param>
+        /// <param name="configureLogger">The delegate for configuring the <see cref="LoggerConfiguration" /> that will be used to construct a <see cref="ILogger" />.</param>
         /// <param name="preserveStaticLogger">Indicates whether to preserve the value of <see cref="Log.Logger"/>.</param>
         /// <param name="writeToProviders">By default, Serilog does not write events to <see cref="ILoggerProvider"/>s registered through
         /// the Microsoft.Extensions.Logging API. Normally, equivalent Serilog sinks are used in place of providers. Specify

--- a/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
+++ b/src/Serilog.AspNetCore/SerilogWebHostBuilderExtensions.cs
@@ -1,4 +1,4 @@
-﻿// Copyright 2017-2019 Serilog Contributors
+// Copyright 2017-2019 Serilog Contributors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -16,6 +16,7 @@ using System;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Logging;
+using Serilog.Core;
 using Serilog.Extensions.Hosting;
 using Serilog.Extensions.Logging;
 

